tests/src/baseservices/threading/coverage/OSThreadId/OSThreadId.cs

```diff
@@ -0,0 +1,90 @@
+using System;
+using System.Diagnostics;
+using System.Reflection;
+using System.Threading;
+
+namespace Threading.Tests
+{
+    public sealed class OSThreadId
+    {
+        private const int NumThreads = 10;
+        private static MethodInfo s_osThreadIdGetMethod;
+        private static ManualResetEvent s_resetEvent = new ManualResetEvent(false);
+        private static ulong[] s_threadIds = new ulong[NumThreads];
+
+        public static int Main(string[] args)
+        {
+            // The property to be tested is internal.
+            Type runtimeThreadType = typeof(object).Assembly.GetType("Internal.Runtime.Augments.RuntimeThread");
+            Debug.Assert(runtimeThreadType != null);
+            PropertyInfo osThreadIdProperty = runtimeThreadType.GetProperty("CurrentOSThreadId", BindingFlags.NonPublic | BindingFlags.Static);
+            Debug.Assert(osThreadIdProperty != null);
+            s_osThreadIdGetMethod = osThreadIdProperty.GetGetMethod(true);
+            Debug.Assert(s_osThreadIdGetMethod != null);
+
+            // Test the main thread.
+            Debug.Assert(GetCurrentThreadId() > 0);
+
+            // Create more threads.
+            Thread[] threads = new Thread[NumThreads];
+            for (int i = 0; i < NumThreads; i++)
+            {
+                threads[i] = new Thread(new ParameterizedThreadStart(ThreadProc));
+                threads[i].Start(i);
+            }
+
+            // Now that all threads have been created, allow them to run.
+            s_resetEvent.Set();
+
+            // Wait for all threads to complete.
+            for (int i = 0; i < NumThreads; i++)
+            {
+                threads[i].Join();
+            }
+
+            // Check for duplicate thread IDs.
+            Array.Sort(s_threadIds);
+            ulong previousThreadId = 0;
+            for (int i = 0; i < NumThreads; i++)
+            {
+                if (i == 0)
+                {
+                    previousThreadId = s_threadIds[i];
+                }
+                else
+                {
+                    Debug.Assert(s_threadIds[i] > 0);
```

Comment:
@brianrob why are you using `Debug.Assert` instead of `Assert`? Should the tested API work only in Debug?

Reply:
@adamsitnik, thanks for pointing this out.  The test should work on all flavors.  I have replaced calls to ```Debug.Assert``` with calls to a test-local ```Assert``` as there isn't a platform-level ```Assert```.